Modules/_xxsubinterpretersmodule.c

```diff
@@ -2323,6 +2353,74 @@ PyDoc_STRVAR(channel_list_all_doc,
 \n\
 Return the list of all IDs for active channels.");
 
+
+static PyObject *
+channel_list_interpreters(PyObject *self, PyObject *args, PyObject *kwds)
+{
+    static char *kwlist[] = {"cid", "send", "recv", NULL};
+    int64_t cid;
+    _PyChannelState *chan;
+    int send = 0;           /* Send end? */
+    int recv = 0;           /* Receive end? */
+    int64_t *ids = NULL;    /* Array of interpreter IDs */
+    int64_t count = 0;      /* Number of interpreters to return */
+    PyObject *ret = NULL;   /* Python list of interpreter IDs */
+
+    if (!PyArg_ParseTupleAndKeywords(
+            args, kwds, "O&|$pp:channel_list_interpreters",
+            kwlist, channel_id_converter, &cid, &send, &recv)) {
+        return NULL;
+    }
+
+    if ((send && recv) || (!send && !recv)) {
+        PyErr_SetString(PyExc_ValueError,
+                        "Specify exactly one of send or recv");
+        goto except;
+    }
+
+    chan = _channels_lookup(&_globals.channels, cid, NULL);
+    if (chan == NULL) {
+        goto except;
+    }
+
+    ids = _channelends_list_interpreters(chan->ends, &count, send);
+    if (ids == NULL) {
+        goto except;
+    }
+
+    ret = PyList_New((Py_ssize_t)count);
```

Comment:
With c99 support (see PEP 7), you don't need to put all the declarations at the top of the function.  It's nice to keep the declaration closer to where it is used, so:

```suggestion
    PyObject *ret = PyList_New((Py_ssize_t)count);
```

Reply:
Fair enough, I don't disagree, this is just a departure from the coding standards we follow at work that strictly enforce all variable declaration at the top of functions :)